executor/executor_write.go

```diff
@@ -407,6 +411,162 @@ func (e *DeleteExec) Close() error {
 	return e.SelectExec.Close()
 }
 
+// NewLoadDataInfo returns a LoadDataInfo structure, and it's only used for tests now.
+func NewLoadDataInfo(row []types.Datum, ctx context.Context, tbl table.Table) *LoadDataInfo {
+	return &LoadDataInfo{
+		row:       row,
+		insertVal: &InsertValues{ctx: ctx, Table: tbl},
+		Table:     tbl,
+	}
+}
+
+// LoadDataInfo saves the information of loading data operation.
+type LoadDataInfo struct {
+	row       []types.Datum
+	insertVal *InsertValues
+
+	Path       string
+	Table      table.Table
+	FieldsInfo *ast.FieldsClause
+	LinesInfo  *ast.LinesClause
+}
+
+// InsertData inserts data into specified table according to the specified format.
+// If it has the rest of data isn't completed the processing, then is returns without completed data.
+func (e *LoadDataInfo) InsertData(data1, data2 []byte) ([]byte, error) {
```

Comment:
I think changing names of "data1/2" to remainedData and appendingData is better :D